File: backend/src/main/java/com/example/stamp_app/repository/MicroControllerRepository.java
package com.example.stamp_app.repository;

import com.example.stamp_app.entity.MicroController;

import java.util.UUID;

import org.springframework.data.jpa.repository.JpaRepository;

public interface MicroControllerRepository extends JpaRepository<MicroController, UUID> {
    MicroController findByMacAddress(String macAddress);

    MicroController findByUuid(UUID uuid);
}

File: backend/src/main/java/com/example/stamp_app/repository/MicroControllerRepository.kt
package com.example.stamp_app.repository

import com.example.stamp_app.entity.MicroController
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface MicroControllerRepository : JpaRepository<MicroController?, UUID?> {

    fun findByMacAddress(macAddress: String): MicroController?

    fun findByUuid(uuid: UUID): MicroController?

}
